src/util/xeval/evaluator.rs

```diff
@@ -225,13 +232,40 @@ impl Evaluator {
         Ok((&children[0], &children[1]))
     }
 
+    fn eval_one_child(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<Datum> {
+        let children = expr.get_children();
+        if children.len() != 1 {
+            return Err(Error::Expr(format!("{:?} need 1 operands but got {}",
+                                           expr.get_tp(),
+                                           children.len())));
+        }
+        let child = try!(self.eval(ctx, &children[0]));
+        Ok(child)
+    }
+
     fn eval_two_children(&mut self, ctx: &EvalContext, expr: &Expr) -> Result<(Datum, Datum)> {
         let (left_expr, right_expr) = try!(self.get_two_children(expr));
         let left = try!(self.eval(ctx, left_expr));
         let right = try!(self.eval(ctx, right_expr));
         Ok((left, right))
     }
 
+    fn eval_more_children(&mut self,
+                          ctx: &EvalContext,
+                          expr: &Expr,
+                          num: usize)
```

Comment:
I'd like to express more than `num` children, any better suggestion? @lishihai9017